### PR TITLE
IGNITE-13426 Adds command to print system view content.

### PR DESCRIPTION
**Pull request by ololo3000**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+    /**
+     * Prints system view content obtained via {@link VisorSystemViewTask} execution.
+     *
+     * @param taskRes Result of {@link VisorSystemViewTask} execution.
+     * @param log Logger.
+     */
+    private void printSystemViewContent(VisorSystemViewTaskResult taskRes, Logger log) {
+        List<String> colTitles = taskRes.systemViewAttributes();
+
```
Comment (xtern) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:106]: I suggest removing this newline

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        List<? extends List<?>> sysViewRows = taskRes.systemViewContent();
```
Comment (xtern) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:107]: ? extends List<?> can be replaced with List<?>

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+     * @param log Logger.
+     */
+    private void printRow(
+        Collection<String> row,
+        Collection<SimpleAttributeType> types,
+        Collection<Integer> lengths,
+        Logger log
+    ) {
+        Iterator<SimpleAttributeType> typeIter = types.iterator();
+
```
Comment (xtern) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:145]: I suggest removing this newline

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+     * @param types Column types in sequential order for decent row formatting.
+     * @param lengths Column lengths in sequential order for decent row formatting.
+     * @param log Logger.
+     */
+    private void printRow(
+        Collection<String> row,
+        Collection<SimpleAttributeType> types,
+        Collection<Integer> lengths,
+        Logger log
+    ) {
```
Comment (xtern) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:143]: we can add an assertion here to ensure code reader that the sizes of collections are equal

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.visor.systemview;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
```
Comment (xtern) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java:24]: Unused import

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                    if (toSqlName(sysView.name()).equals(targetSysViewName)) {
+                        targetSysView = sysView;
+
+                        break;
+                    }
+                }
+            }
+
+            if (targetSysView == null)
+                return null;
```
Comment (xtern) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java:94]: I advise you to separate the logic for finding a view into a separate method.
In this case, code here will look something like 
```
SystemView<?> targetSysView = findSysView(arg.systemViewName());

            if (targetSysView == null)
                return null;
```

`findSysView` may look something like

```
private SystemView<?> findSysView(String name) {
            GridSystemViewManager sysViewMgr = ignite.context().systemView();

            SystemView<?> targetSysView = sysViewMgr.view(name);

            if (targetSysView != null)
                return targetSysView;

            // comment about sql...
            for (SystemView<?> sysView : sysViewMgr) {
                if (toSqlName(sysView.name()).equals(name))
                    return sysView;
            }

            return null;
        }
```

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+            List<String> attrNames = new ArrayList<>();
+
```
Comment (xtern) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java:97]: I suggest removing this newline

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+/**
+ * Reperesents result of {@link VisorSystemViewTask}.
+ */
+public class VisorSystemViewTaskResult extends IgniteDataTransferObject {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** Attribute values for each row of the system view. */
+    private List<? extends List<?>> rows;
```
Comment (xtern) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java:37]: `List<? extends List<?>>` can be replaced with `List<List<?>>`

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java (hunk 1 above)
Comment (ololo3000) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java]: Done.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java (hunk 2 above)
Comment (ololo3000) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java]: Done.

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java (hunk 3 above)
Comment (ololo3000) [reply modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java (hunk 5 above)
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java (hunk 6 above)
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java:79]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java (hunk 7 above)
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTask.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java (hunk 8 above)
Comment (ololo3000) [reply modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java]: Done.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * Default constructor.
+     */
+    public VisorSystemViewTaskResult() {
+        // No-op.
+    }
+
+    /**
+     * @param attrs Names of system view attributes.
+     * @param rows Attribute values for each row of the system view.
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java:54]: nit: @param types missed in javadoc.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+     */
+    public VisorSystemViewTaskResult(List<String> attrs, List<SimpleAttributeType> types, List<List<?>> rows) {
+        this.attrs = attrs;
+        this.types = types;
+        this.rows = rows;
+    }
+
+    /**
+     * @return Names of the system view attributes.
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java:63]: Let's make this and similar comment one liner.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+    /**
+     * @return Types of the system view attributes.
+     */
+    public List<SimpleAttributeType> systemViewAttributeTypes() {
+        return types;
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java:84]: Please, make it one line comment.

File: modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    public VisorSystemViewTaskResult(List<String> attrs, List<SimpleAttributeType> types, List<List<?>> rows) {
+        this.attrs = attrs;
+        this.types = types;
+        this.rows = rows;
+    }
+
+    /**
+     * @return Names of the system view attributes.
+     */
+    public List<String> systemViewAttributes() {
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/visor/systemview/VisorSystemViewTaskResult.java:65]: Let's make method naming consistent with the variables names.

attrs -> attributes
rows -> rows
types -> types

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+    /**
+     * Prints system view content obtained via {@link VisorSystemViewTask} execution.
+     *
+     * @param taskRes Result of {@link VisorSystemViewTask} execution.
+     * @param log Logger.
+     */
+    private void printSystemViewContent(VisorSystemViewTaskResult taskRes, Logger log) {
+        List<String> colTitles = taskRes.systemViewAttributes();
```
Comment (nizhikov) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:105]: Let's rename this to `titles`

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+        List<List<?>> sysViewRows = taskRes.systemViewContent();
```
Comment (nizhikov) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:106]: Let's rename this to `rows`

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+
+        List<Integer> colLenghts = colTitles.stream().map(String::length).collect(Collectors.toList());
```
Comment (nizhikov) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:108]: Let's rename this to `colSzs`

File: modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java [hunk 16]
```diff
@@ -0,0 +1,221 @@
...
+
+        List<List<String>> rows = new ArrayList<>(sysViewRows.size());
+
+        sysViewRows.forEach(sysViewRow -> {
+            ListIterator<Integer> colLenIter = colLenghts.listIterator();
+
+            rows.add(sysViewRow.stream().map(attr -> {
+                String colVal = String.valueOf(attr);
```
Comment (nizhikov) [inline modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/systemview/SystemViewCommand.java:116]: Let's rename this to `val`

**Review by nizhikov — APPROVED** [review]
I left some minor comments. Please fix them
LGTM

**Review by nizhikov — APPROVED** [review]

**Pull request merged**